src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -41,13 +44,21 @@ public DnxCatalogCollector(Uri index, StorageFactory storageFactory, Func<HttpMe
                     string nuspec = await LoadNuspec(client, id, version, cancellationToken);
                     if (nuspec != null)
                     {
-                        var requestUri = Utilities.GetNugetCacheBustingUri(new Uri(ContentBaseAddress, string.Format("{0}.{1}.nupkg", id, version)));
-                        using (Stream stream = await client.GetStreamAsync(requestUri))
+                        Uri sourceUri= new Uri(ContentBaseAddress, string.Format("{0}.{1}.nupkg", id, version));
+                        var requestUri = Utilities.GetNugetCacheBustingUri(sourceUri);
+                        if (await storage.IsUpdateRequiredAsync(sourceUri, id, version))
                         {
-                            await _dnxMaker.AddPackage(stream, nuspec, id, version, cancellationToken);
-                        }
+                            using (Stream stream = await client.GetStreamAsync(requestUri))
```

Comment:
`GetStreamAsync` should also determine `ExistsAsync` and the `Content-MD5` header. We don't need this many HTTP requests.

Reply:
As we discussed offline I will add a task to track the reduction of the http requests. The current change will add one more in case that changes are required.

Reply:
https://github.com/NuGet/Engineering/issues/1026

Reply:
Agree with the below.